#!/bin/bash
#$ -cwd
#Abort on any error,
set -e


cd /PATH_TO/Mp_O/host_nonhost/


STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_24h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_5_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_24h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_4_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_24h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_24h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_24h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_24h_1_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_3h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_5_R2.fq.gz

######

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_3h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_4_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_3h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_3h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_3h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_3h_1_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_24h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_5_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_24h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_4_R2.fq.gz

#######

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_24h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_24h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_host_24h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_host_24h_1_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_3h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_3h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_3h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_1_R2.fq.gz


##################################
STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_3h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_5_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_art_3h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_art_3h_4_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_24h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_24h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_24h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_1_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_24h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_5_R2.fq.gz

#####

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_24h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_24h_4_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_3h_5 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_5_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_5_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_3h_4 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_4_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_4_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_3h_3 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_3_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_3_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_3h_2 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_2_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_2_R2.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_nonhost_3h_1 --readFilesIn /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_1_R1.fq.gz /mnt/shared/projects/aphids/20151212_Host_nonhost_artificaial_RNAseq/trimmed_files/final_files/Mp_nonhost_3h_1_R2.fq.gz


####### heads vs bodies




STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_body1 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5523_LDI4454/809_LIB5523_LDI4454_TGACCA_L007_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5523_LDI4454/809_LIB5523_LDI4454_TGACCA_L007_R2_paired.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_body2 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5515_LDI4455/809_LIB5515_LDI4455_ACAGTG_L007_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5515_LDI4455/809_LIB5515_LDI4455_ACAGTG_L007_R2_paired.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_body3 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5516_LDI4456/809_LIB5516_LDI4456_GCCAAT_L007_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5516_LDI4456/809_LIB5516_LDI4456_GCCAAT_L007_R2_paired.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_head1 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5513_LDI4451/809_LIB5513_LDI4451_CGTACG_L008_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5513_LDI4451/809_LIB5513_LDI4451_CGTACG_L008_R2_paired.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_head2 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB4709_LDI4452/809_LIB4709_LDI4452_GAGTGG_L008_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB4709_LDI4452/809_LIB4709_LDI4452_GAGTGG_L008_R2_paired.fq.gz

STAR --genomeDir /PATH_TO/Mp_O/Mp_O_host_non_RNAseq_assembly/star_indicies/ --runThreadN 10 --sjdbGTFfile Mp_O_v1.gtf --quantMode TranscriptomeSAM --outSAMmapqUnique 255 --outSAMtype BAM SortedByCoordinate --outFilterMismatchNmax 3 --readFilesCommand zcat --outFileNamePrefix /PATH_TO/Mp_O/host_nonhost/Mp_O_head3 --readFilesIn /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5514_LDI4453/809_LIB5514_LDI4453_CGATGT_L007_R1_paired.fq.gz /mnt/shared/projects/aphids/201311_Aphid_RNAseq/Aphid_RNAseq_repeated_data/PKG_Re_run_Lanes_1_2/Sample_809_LIB5514_LDI4453/809_LIB5514_LDI4453_CGATGT_L007_R2_paired.fq.gz



